### PR TITLE
FYLE-2cqxfzg-1: Unlink corporate card expenses

### PR DESCRIPTION
**Pull request by Dimple16**
### Added currency in card amount section and updated information box content
Screenshot 2022-05-30 at 4 35 07 PM

### Added Unlink card details Option
Screenshot 2022-05-30 at 4 37 56 PM

### Confirmation pop up to unlink details
Screenshot 2022-05-30 at 3 51 37 PM

### Unlink success information
Screenshot 2022-05-30 at 3 52 01 PM

### Expense history section updated with unlink card section
Screenshot 2022-05-30 at 5 05 46 PM

### CODE REVIEW

**Comment by Dimple16** [comment]
@rvab @abhishek1234321 These changes were previously approved and merged here - https://github.com/fylein/fyle-mobile-app/pull/1109
I reverted the change as there were new updated backend side - made those changes and created a new pull request

File: src/app/core/services/status.service.ts [hunk 1]
```diff
@@ -197,6 +197,12 @@ export class StatusService {
           icon: 'duplicate',
         };
         break;
+      case lowerCaseComment.indexOf('card details unlinked') > -1:
```
Comment (rvab) [inline src/app/core/services/status.service.ts:200]: use `includes` instead of `indexOf`

File: src/app/core/services/status.service.ts [hunk 2]
```diff
@@ -197,6 +197,12 @@ export class StatusService {
...
+        statusCategory = {
+          category: 'Card details unlinked',
+          icon: 'fy-corporate-card',
+        };
```
Comment (rvab) [inline src/app/core/services/status.service.ts:204]: Can you define an interface for `statusCategory` or create an object with predefined keys? For e.g: 
```
let statusCategory = {
  category: '',
  icon: ''
}
```
This helps in avoiding mistyping keys. Not a blocker. We should start using types as much as possible.

File: src/app/fyle/add-edit-expense/add-edit-expense.page.ts [hunk 3]
```diff
@@ -2840,6 +2912,9 @@ export class AddEditExpensePage implements OnInit {
       this.isCccExpense = etxn?.tx?.corporate_credit_card_expense_group_id;
       this.isExpenseMatchedForDebitCCCE = !!etxn?.tx?.corporate_credit_card_expense_group_id && etxn.tx.amount > 0;
       this.canDismissCCCE = !!etxn?.tx?.corporate_credit_card_expense_group_id && etxn.tx.amount < 0;
+      this.canUnlinkCCCE =
+        !!etxn?.tx?.corporate_credit_card_expense_group_id &&
+        ['APPROVER_PENDING', 'COMPLETE', 'DRAFT'].includes(etxn?.tx?.state);
```
Comment (rvab) [inline src/app/fyle/add-edit-expense/add-edit-expense.page.ts:2917]: Can we create an enum for `etxn.txn.state`?

File: src/app/fyle/add-edit-expense/add-edit-expense.page.ts [hunk 4]
```diff
@@ -2820,6 +2823,7 @@ export class AddEditExpensePage implements OnInit {
           this.matchedCCCTransaction = matchedExpense[0].ccce;
           this.matchingCCCTransactions = matchingTransactions;
           this.selectedCCCTransaction = this.matchedCCCTransaction;
+          this.cardCurrencySymbol = getCurrencySymbol(this.selectedCCCTransaction?.currency, 'wide');
```
Comment (rvab) [inline src/app/fyle/add-edit-expense/add-edit-expense.page.ts:2826]: What does `cardCurrencySymbol` do?

**Review by rvab — CHANGES_REQUESTED** [review]

File: src/app/core/services/status.service.ts (hunk 1 above)
Comment (Dimple16) [reply src/app/core/services/status.service.ts:200]: not being used anymore, removed this section

File: src/app/fyle/add-edit-expense/add-edit-expense.page.ts (hunk 4 above)
Comment (Dimple16) [reply src/app/fyle/add-edit-expense/add-edit-expense.page.ts:2826]: shows the currency symbol `$` in the amount $430 
as we don't use `USD` directly

File: src/app/fyle/add-edit-expense/add-edit-expense.page.ts (hunk 3 above)
Comment (Dimple16) [reply src/app/fyle/add-edit-expense/add-edit-expense.page.ts:2917]: please let me know if this is required? we are using this only here though,

Comment (rvab) [reply src/app/fyle/add-edit-expense/add-edit-expense.page.ts:2917]: I would suggest creating an enum for `state`. Can take this up later

**Review by rvab — APPROVED** [review]

File: src/app/fyle/add-edit-expense/add-edit-expense.page.html [hunk 5]
```diff
@@ -137,15 +137,15 @@
                 <div class="add-edit-expense--card-container__label">Spend Amount</div>
                 <div *ngIf="selectedCCCTransaction?.amount > 0">
                   <span class="add-edit-expense--card-container__value"
-                    >{{ selectedCCCTransaction?.amount | humanizeCurrency: selectedCCCTransaction?.currency:2:true
-                    }}</span
+                    >{{ cardCurrencySymbol }}{{ selectedCCCTransaction?.amount | humanizeCurrency:
+                    selectedCCCTransaction?.currency:2:true }}</span
```
Comment (rvab) [inline src/app/fyle/add-edit-expense/add-edit-expense.page.html:141]: I'm assuming this will be displayed as `$10.23` instead of `$10.23 USD`

Comment (Dimple16) [reply src/app/fyle/add-edit-expense/add-edit-expense.page.html:141]: yes

**Review by abhishek1234321 — APPROVED** [review]

**Pull request merged**